### ISSUE TITLE
EarlyStopping mode-parameter "auto" is missleading

### ISSUE THREAD

**Issue by Lennart4711** [keras-team/tf-keras#253]
In the [docs on EarlyStopping](https://keras.io/api/callbacks/early_stopping/) it says: "in 'auto' mode, the direction is automatically inferred from the name of the monitored quantity." suggesting there is some sort of "smart" way/ lookup table for different metrics. When you look at the  [corresponding code](https://github.com/keras-team/keras/blob/e6784e4302c7b8cd116b74a784f4b78d60e83c26/keras/callbacks.py#L2038) however, you see that it is just a string-comparision wether the name of the monitored variable ends in acc, accuracy or auc. 
```python
if (
    self.monitor.endswith("acc")
    or self.monitor.endswith("accuracy")
    or self.monitor.endswith("auc")
):
    self.monitor_op = np.greater
else:
    self.monitor_op = np.less
```
I believe this is missleading and causes confusion. 
You can see someone struggeling with it in [this stackoverflow post](https://stackoverflow.com/questions/69744405/tensorflow-earlystopping-stops-too-early) who thinks that "TF has the direction of ndcg (Normalized Discounted Cumulative Gain) for ranking problems wrong." when in reality it does not have direction for anything.

I don't know the consequences of changing this, as someone might have choosen the right value by accident and changing the standard value could break their pipeline(s). I do **not** have a fix because I don't know how widespread this problem is. I just wanted to let you know that this could be an issue worth looking into.
